src/containers/LoginFormContainer.js

```diff
@@ -1,114 +1,69 @@
-import axios from "axios";
-import { Redirect } from "react-router-dom";
-import HomePage from "../components/HomePage/HomePage";
-import EditProfileContainer from "./EditProfileContainer";
+import { API_FB_ID, FACEBOOK_SCOPES, API_GOOGLE_ID } from "../constants";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import {
+  loginRequested,
+  handleGoogleSuccess,
+  handleGoogleFailure,
+  handleFacebookResponse
+} from "../actions/LoginActions";
+import LoginForm from "../components/Login/LoginForm";
 
-export default class LoginFormContainer extends Component {
-  state = {
-    isLoggedIn: false,
-    userID: "",
-    firstName: "",
-    lastName: "",
-    email: "",
-    photoUrl: "",
-    externalProviderToken: "",
-    roleName: "",
+class LoginFormContainer extends Component {
+  loginRequested = () => {
+    this.props.actions.loginRequested();
   };
 
-  componentClicked = () => console.log("clicked");
-
-  responseFacebook = response => {
-    console.log(response);
-
-    this.setState({
-      isLoggedIn: true,
-      userID: response.userID,
-      firstName: response.first_name,
-      lastName: response.last_name,
-      email: response.email,
-      photoUrl: response.picture.data.url,
-      externalProviderToken: response.accessToken,
-      externalProviderName: providers.FACEBOOK,
-      roleName: roles.USER
-    });
-
-    this.addNewUser();
-    console.log(response.accessToken);
+  handleFacebookResponse = response => {
+    this.props.actions.handleFacebookResponse(response);
   };
 
-  responseGoogle = response => {
-    console.log(response);
-
-    this.setState({
-      isLoggedIn: true,
-      externalProviderToken: response.accessToken,
-      firstName: response.profileObj.givenName,
-      lastName: response.profileObj.familyName,
-      email: response.profileObj.email,
-      birthday: response.birthday,
-      photoUrl: response.profileObj.imageUrl,
-      externalProviderName: providers.GOOGLE,
-      roleName: roles.USER
-
-    });
-    this.addNewUser();
+  handleGoogleFailure = response => {
+    this.props.actions.handleGoogleFailure(response);
   };
 
-  addNewUser = () => {
-    const { firstName, lastName, email, photoUrl, externalProviderToken, externalProviderName, roleName } = this.state;
-    axios.post(`${BASE_URL}/login`, {
-        firstName: firstName,
-        lastName: lastName,
-        email: email,
-        photoUrl: photoUrl,
-        externalProviderToken: externalProviderToken,
-        externalProviderName: externalProviderName,
-        roleName: roleName
-
-      })
-      .then((response) => {
-        if(response.status === 200)
-          this.props.history.push(HomePage);
-        else if(response.status === 201)
-          this.props.history.push(EditProfileContainer);
-        else
-          console.log("Something went wrong...");
-      })
-      .catch(err => console.log(err));
+  handleGoogleSuccess = response => {
```

Comment:
I'm not sure if it's needed, you can use this.props.handleGoogleSuccess just inside render, but you would have to change  mapDispatchToProps not to use actions there

Reply:
I wanted to explicitly separate redux from rendering, not sure if it's correct @jskurzynska